src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1878,11 +1878,14 @@ public async Task PostAsync_CallMethod_EmptyContent(Uri remoteServer)
         [InlineData(false, "1.1")]
         [InlineData(true, "1.1")]
         [InlineData(null, "1.1")]
+        [InlineData(false, "2.0")]
+        [InlineData(true, "2.0")]
+        [InlineData(null, "2.0")]
         public async Task PostAsync_ExpectContinue_Success(bool? expectContinue, string version)
```

Comment:
I don't understand this test.  Why is the parameter for 'expectContinue' a nullable-bool?

Reply:
I can look into it. This is existing test I updated for 2.0

Reply:
> Why is the parameter for 'expectContinue' a nullable-bool?

To validate the behavior when you explicitly set it to true (expectContinue==true), explicitly set it to false (expectContinue==false), or don't set it to anything explicitly and just have the default (expectContinue==null).

Reply:
I still don't really understand this.  A request either sets the 'Expect: 100-continue' header or it doesn't set the header at all.  Are you saying the test tries to validate cases where the header is 'Expect: somethingelse' ?

Reply:
Seems like ExpectContinue is nullable bool?
So this seems to verify default behavior when is it is not explicitly set.

Reply:
> So this seems to verify default behavior when is it is not explicitly set.

Exactly.  Did the developer explicitly set the ExpectContinue header to true, did they explicitly set the ExpectContinue header to false, or did they not touch it.

Reply:
I guess I'm confused by this because I thought the developer added the header manually.  It sounds like you think they call a property on the HttpRequestMessage that is a bool?

Reply:
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs#L69

Reply:
Ok thanks for the reminders of how this property/header works.

In hindsight, this property should probably have been a normal bool.  Where true' means to emit the 'Expect: 100-continue' request header and 'false' means to emit no header.  The RFC doesn't define any other value for the 'Expect' header.